### ISSUE TITLE
passing results of work to next scheduled work

### ISSUE THREAD

**Issue by aaur0** [j-easy/easy-flows#7]
I have a sequential workflow where the result of first work has to be passed to the next. I was wondering if it can be achieved using easy-flows

**Comment by zhhaojie** [comment]
+1 .
i am also thinking there are some bugs of `Work & WorkReport` 's design.
`Work` extends Callable<V> .  it should return somethings, and now is `WorkReport`.
but `WorkReport` without the real result of Callable!

**Comment by benas** [comment]
@aaur0 you can do that by sharing an object (result holder) between steps. Otherwise, you can extend `DefaultWorkReport` to add the result and create a custom sequential flow that passes the result between work units. I would keep it simple and use a shared result holder object between steps.

@zhhaojie Can you elaborate please? Let's assume we add the result to the report, how would next work units consume it? Do you mean we need to change the `Work` interface to accept the "previous" report?

**Comment by benas** [comment]
It looks like this is similar to #2 where I gave an example of sharing an execution context between work units: https://gist.github.com/benas/4d55e8b18f2ba05ab94a14eda98bd6e9.

**Comment by benas** [comment]
I believe there is an added value in introducing an execution context concept in Easy Flows in order to free the user from creating it manually and share it between all work units at construction time (as shown in my previous example).

So I introduced a new concept called [`WorkContext`](https://github.com/j-easy/easy-flows/blob/master/src/main/java/org/jeasy/flows/work/WorkContext.java) which makes it possible to:

1. Pass initial parameters to the workflow 
2. Share data between sequential/parallel/conditional work units
3. Implement a stateful repeat flow

An example can be found [here](https://github.com/j-easy/easy-flows/blob/964a7c588ece95903de982616666cde46f541d72/src/test/java/org/jeasy/flows/engine/WorkFlowEngineImplTest.java#L142). This is basically the classical word count example: each work unit executed in parallel calculates the number of words in a given text partition and puts its partial count in the execution context (which is thread-safe). Then another unit of work aggregates the results from the execution context. Partitions are passed as initial parameters to the workflow. This example shows how the execution context can be used to pass initial parameters to the workflow and share data between sequential and parallel work units.

You can give it a try in version 0.2-SNAPSHOT (See [here](https://github.com/j-easy/easy-flows/wiki/getting-started#use-a-snapshot-version) if needed). 

@aaur0 @zhhaojie Looking forward for your feedback!

**Comment by benas** [comment]
> I have a sequential workflow where the result of first work has to be passed to the next. I was wondering if it can be achieved using easy-flows

@aaur0 v0.2 has been released with the introduction of `WorkContext` which allows you to share data between steps as shown in the previous comment.

> i am also thinking there are some bugs of Work & WorkReport 's design.
Work extends Callable . it should return somethings, and now is WorkReport.
but WorkReport without the real result of Callable .

@zhhaojie Thank you for the feedback on this. I fixed the design by introducing the `WorkContext` concept as explained previously.
